cli/pkg/validate/required_keys.go

```diff
@@ -92,5 +112,247 @@ func (rk *RequiredKeysValidator) Validate(dc *catalog.DataCatalog) []ValidationE
 		}
 	}
 
+	// Custom Types required keys
+	for group, customTypes := range dc.CustomTypes {
+		for _, customType := range customTypes {
+			reference := fmt.Sprintf("#/custom-types/%s/%s", group, customType.LocalID)
+
+			// Check mandatory fields
+			if customType.LocalID == "" || customType.Name == "" || customType.Type == "" {
+				errors = append(errors, ValidationError{
+					error:     fmt.Errorf("id, name and type fields on custom type are mandatory"),
+					Reference: reference,
+				})
+				continue
+			}
+
+			// Check if config is present
+			if customType.Config == nil {
+				errors = append(errors, ValidationError{
+					error:     fmt.Errorf("config field is mandatory on custom type"),
+					Reference: reference,
+				})
+			}
+
+			// If type is object, properties is required
+			if customType.Type == "object" && len(customType.Properties) == 0 {
+				errors = append(errors, ValidationError{
+					error:     fmt.Errorf("properties array is required for custom type with type 'object'"),
+					Reference: reference,
+				})
+			}
+
+			// Check each property in properties has id field
+			if customType.Type == "object" {
+				for i, prop := range customType.Properties {
+					if prop.ID == "" {
+						errors = append(errors, ValidationError{
+							error:     fmt.Errorf("id field is mandatory for property at index %d in custom type", i),
+							Reference: reference,
+						})
+					}
+				}
+			}
```

Comment:
This is going to be replaced with a check for "$ref", right?

Reply:
Yes